components/launcher/src/server/mod.rs

```diff
@@ -450,13 +454,52 @@ fn dispatch(tx: &Sender, bytes: &[u8], services: &mut ServiceTable) {
 }
 
 fn setup_connection(server: IpcOneShotServer<Vec<u8>>) -> Result<(Receiver, Sender)> {
-    let (rx, raw) = server.accept().map_err(|_| Error::AcceptConn)?;
-    let txn = protocol::NetTxn::from_bytes(&raw).map_err(Error::Deserialize)?;
-    let mut msg = txn.decode::<protocol::Register>()
-        .map_err(Error::Deserialize)?;
-    let tx = IpcSender::connect(msg.take_pipe()).map_err(Error::Connect)?;
-    send(&tx, &protocol::NetOk::new())?;
-    Ok((rx, tx))
+    let pair = Arc::new((Mutex::new(false), Condvar::new()));
```

Comment:
I suggest viewing this part with [whitespace changes suppressed](https://github.com/habitat-sh/habitat/pull/5393/files?&w=1), so you can see that the core logic hasn't changed at all.